File: solutions_5648941810974720_1/java/SapFire/GettingTheDigits.java
import java.util.Scanner;
public class GettingTheDigits {

    public static String found(String s, String t) {
        for (int i = 0; i < s.length(); i++) {
            if (s.substring(i, i + 1).equals(t)) {
                return s.substring(0, i) + s.substring(i + 1);
            }
        }
        return "-1";
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for (int i = 0; i < t; i++) {
            String s = in.next();
            int[] digits = new int[10];
            for (int j = 0; j < digits.length; j++) {
                digits[j] = 0;
            }
            boolean zero = false;
            while (true) {
                zero = false;
                String str = found(s, "Z");
                if (!str.equals("-1")) {
                    s = found(s, "Z");
                    digits[0]++;
                    s = found(s, "E");
                    s = found(s, "R");
                    s = found(s, "O");
                    zero = true;
                }
                if (!zero) {
                    break;
                }
            }
            boolean six = false;
            while (true) {
                six = false;
                String str = found(s, "S");
                String s2 = found(s, "I");
                String s3 = found(s, "X");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1")) {
                    s = found(s, "S");
                    digits[6]++;
                    s = found(s, "I");
                    s = found(s, "X");
                    six = true;
                }
                if (!six) {
                    break;
                }
            }
            boolean two = false;
            while (true) {
                two = false;
                String str = found(s, "T");
                String s2 = found(s, "W");
                String s3 = found(s, "O");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1")) {
                    s = found(s, "T");
                    digits[2]++;
                    s = found(s, "W");
                    s = found(s, "O");
                    two = true;
                }
                if (!two) {
                    break;
                }
            }
            boolean eight = false;
            while (true) {
                eight = false;
                String str = found(s, "E");
                String s2 = found(s, "I");
                String s3 = found(s, "G");
                String s4 = found(s, "H");
                String s5 = found(s, "T");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1") && !s4.equals("-1") && !s5.equals("-1")) {
                    s = found(s, "E");
                    digits[8]++;
                    s = found(s, "I");
                    s = found(s, "G");
                    s = found(s, "H");
                    s = found(s, "T");
                    eight = true;
                }
                if (!eight) {
                    break;
                }
            }
            boolean four = false;
            while (true) {
                four = false;
                String str = found(s, "F");
                String s2 = found(s, "O");
                String s3 = found(s, "U");
                String s4 = found(s, "R");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1") && !s4.equals("-1")) {
                    s = found(s, "F");
                    digits[4]++;
                    s = found(s, "O");
                    s = found(s, "U");
                    s = found(s, "R");
                    four = true;
                }
                if (!four) {
                    break;
                }
            }
            boolean one = false;
            while (true) {
                one = false;
                String str = found(s, "O");
                String s2 = found(s, "N");
                String s3 = found(s, "E");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1")) {
                    s = found(s, "O");
                    digits[1]++;
                    s = found(s, "N");
                    s = found(s, "E");
                    one = true;
                }
                if (!one) {
                    break;
                }
            }
            boolean three = false;
            while (true) {
                three = false;
                String str = found(s, "T");
                String s2 = found(s, "H");
                String s3 = found(s, "R");
                String s4 = found(s, "E");
                String s5 = found(s, "E");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1") && !s4.equals("-1") && !s5.equals("-1")) {
                    s = found(s, "T");
                    digits[3]++;
                    s = found(s, "H");
                    s = found(s, "R");
                    s = found(s, "E");
                    s = found(s, "E");
                    three = true;
                }
                if (!three) {
                    break;
                }
            }
            boolean five = false;
            while (true) {
                five = false;
                String str = found(s, "F");
                String s2 = found(s, "I");
                String s3 = found(s, "V");
                String s4 = found(s, "E");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1") && !s4.equals("-1")) {
                    s = found(s, "F");
                    digits[5]++;
                    s = found(s, "I");
                    s = found(s, "V");
                    s = found(s, "E");
                    five = true;
                }
                if (!five) {
                    break;
                }
            }
            boolean seven = false;
            while (true) {
                seven = false;
                String str = found(s, "S");
                String s2 = found(s, "E");
                String s3 = found(s, "V");
                String s4 = found(s, "E");
                String s5 = found(s, "N");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1") && !s4.equals("-1") && !s5.equals("-1")) {
                    s = found(s, "S");
                    digits[7]++;
                    s = found(s, "E");
                    s = found(s, "V");
                    s = found(s, "E");
                    s = found(s, "N");
                    seven = true;
                }
                if (!seven) {
                    break;
                }
            }
            boolean nine = false;
            while (true) {
                nine = false;
                String str = found(s, "N");
                String s2 = found(s, "I");
                String s3 = found(s, "N");
                String s4 = found(s, "E");
                if (!str.equals("-1") && !s2.equals("-1") && !s3.equals("-1") && !s4.equals("-1")) {
                    s = found(s, "N");
                    digits[9]++;
                    s = found(s, "I");
                    s = found(s, "N");
                    s = found(s, "E");
                    nine = true;
                }
                if (!nine) {
                    break;
                }
            }
            String number = "";
            for (int j = 0; j < digits.length; j++) {
                for (int k = 0; k < digits[j]; k++) {
                    number += j;
                }
            }
            System.out.println("Case #" + (i + 1) + ": " + number);
        
        }
    }
    
}

File: solutions_5648941810974720_1/C/mroschke/getting_the_digits.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

#define MAX_LENGTH 2048

// -- UTILITY

// get the next integer input
int next_int() {
  int ret = 0;
  scanf("%d", &ret);
  return ret;
}

// get the next string input
void next_string(char* out) {
  scanf("%s", out);
}

// -- REAL FUNCTIONS

// instance solver
void solve_instance(char* str) {
  // get length of the string
  int len = strlen(str);

  // record the digits to output
  int digits[10];
  for(int i = 0; i < 10; i++) {
    digits[i] = 0;
  }

  // convert to counts of each digit
  int letters[26];
  for(int i = 0; i < 26; i++) {
    letters[i] = 0;
  }
  for(int i = 0; i < len; i++) {
    letters[tolower(str[i]) - 'a']++;
  }

  // check for the simple cases
  // Z = ZERO
  while(letters['z' - 'a'] > 0) {
    digits[0]++;
    letters['z' - 'a']--;
    letters['e' - 'a']--;
    letters['r' - 'a']--;
    letters['o' - 'a']--;
    len -= 4;
  }

  // W = TWO
  while(letters['w' - 'a'] > 0) {
    digits[2]++;
    letters['t' - 'a']--;
    letters['w' - 'a']--;
    letters['o' - 'a']--;
    len -= 3;
  }

  // G = EIGHT
  while(letters['g' - 'a'] > 0) {
    digits[8]++;
    letters['e' - 'a']--;
    letters['i' - 'a']--;
    letters['g' - 'a']--;
    letters['h' - 'a']--;
    letters['t' - 'a']--;
    len -= 5;
  }

  // X = SIX
  while(letters['x' - 'a'] > 0) {
    digits[6]++;
    letters['s' - 'a']--;
    letters['i' - 'a']--;
    letters['x' - 'a']--;
    len -= 3;
  }
  
  // NOW, S = SEVEN
  while(letters['s' - 'a'] > 0) {
    digits[7]++;
    letters['s' - 'a']--;
    letters['e' - 'a']--;
    letters['v' - 'a']--;
    letters['e' - 'a']--;
    letters['n' - 'a']--;
    len -= 5;
  }

  // NOW V = FIVE
  while(letters['v' - 'a'] > 0) {
    digits[5]++;
    letters['f' - 'a']--;
    letters['i' - 'a']--;
    letters['v' - 'a']--;
    letters['e' - 'a']--;
    len -= 4;
  }

  // now f = four
  while(letters['f' - 'a'] > 0) {
    digits[4]++;
    letters['f' - 'a']--;
    letters['o' - 'a']--;
    letters['u' - 'a']--;
    letters['r' - 'a']--;
    len -= 4;
  }

  // now t = three
  while(letters['t' - 'a'] > 0) {
    digits[3]++;
    letters['t' - 'a']--;
    letters['h' - 'a']--;
    letters['r' - 'a']--;
    letters['e' - 'a']--;
    letters['e' - 'a']--;
    len -= 5;
  }

  // now o = one
  while(letters['o' - 'a'] > 0) {
    digits[1]++;
    letters['o' - 'a']--;
    letters['n' - 'a']--;
    letters['e' - 'a']--;
    len -= 3;
  }

  // remainder is nines
  while(letters['n' - 'a'] > 0) {
    digits[9]++;
    letters['n' - 'a']--;
    letters['i' - 'a']--;
    letters['n' - 'a']--;
    letters['e' - 'a']--;
    len -= 4;
  }

  // sanity checks
  if(len != 0) {
    fprintf(stderr, "WARNING :: len failure\n");
  }
  for(int i = 0; i < 26; i++) {
    if(letters[i] != 0) {
      fprintf(stderr, "Warning :: letter %c nonzero count\n", 'a' + i);
    }
  }

  // print out the digits
  for(int i = 0; i < 10; i++) {
    for(int j = 0; j < digits[i]; j++) {
      printf("%d", i);
    }
  }
}

// main function
int main(int argc, char* argv[]) {
  // number of test cases
  int t = next_int();

  // storage for strings
  char str[MAX_LENGTH];

  // iterate over test cases
  for(int i = 1; i <= t; i++) {
    // read input string
    next_string(str);

    // print prefix
    printf("Case #%d: ", i);

    // solve instance
    solve_instance(str);

    // print suffix
    printf("\n");
  }
}

